option/types.go

```diff
@@ -85,25 +85,20 @@ func (v NetworkList) Build() []string {
 type Listable[T comparable] []T
 
 func (l Listable[T]) MarshalJSON() ([]byte, error) {
-	arrayList := []T(l)
-	if len(arrayList) == 1 {
-		return json.Marshal(arrayList[0])
+	if len(l) == 1 {
+		return json.Marshal(l[0])
 	}
-	return json.Marshal(arrayList)
+	return json.Marshal([]T(l))
 }
 
-func (l *Listable[T]) UnmarshalJSON(content []byte) error {
-	err := json.Unmarshal(content, (*[]T)(l))
-	if err == nil {
-		return nil
-	}
-	var singleItem T
-	newError := json.Unmarshal(content, &singleItem)
-	if newError != nil {
-		return E.Errors(err, newError)
+func (l *Listable[T]) UnmarshalJSON(content []byte) (err error) {
+	if len(content) > 0 && content[0] != '[' {
```

Comment:
more see https://www.json.org/json-en.html

Reply:
![CleanShot 2023-10-08 at 11 31 01@2x](https://github.com/SagerNet/sing-box/assets/3842474/0ac21f7a-796e-44c0-9ac6-a0a144dcf483)

from https://www.ecma-international.org/wp-content/uploads/ECMA-404_2nd_edition_december_2017.pdf

Reply:
what if the content is "null"?